M2/Macaulay2/packages/SCMAlgebras.m2

```diff
@@ -0,0 +1,394 @@
+-- -*- coding: utf-8 -*-
+
+newPackage(
+    "SCMAlgebras",
+    Version => "1.0",
+    Date => "October 2, 2024",
+    Authors => {{Name => "Ernesto Lax", Email => "erlax@unime.it", HomePage => "https://www.researchgate.net/profile/Ernesto-Lax"}},
+    Headline => "a package to work with sequentially Cohen-Macaulay modules or ideals",
+    Keywords => {"Commutative Algebra"},
+    PackageExports => {"Depth","MinimalPrimes"},
+    DebuggingMode => false
+)
+
+export{
+    -- Methods
+    "deficiencyModule",
+    "canonicalModule",
+    "filterIdeal",
+    "isSCM",
+    -- Service
+    "minDim"
+};
+
+
+MyDoc := ""; 
+MyDoc = concatenate(MyDoc,///
+Node
+  Key
+    SCMAlgebras
+  Headline
+    a package to work with sequentially Cohen-Macaulay modules or ideals
+  Description
+    Text
+      SCMAlgebras is a package to check whether a module or an ideal is sequentially Cohen-Macaulay,
+      by computing the modules of deficiency and the filter ideals.
+///);
+
+
+
+-------------
+-- FUNCTIONS |
+---------------------------------------------------------------------------------------------
+--=======================================================================
+-- computes the ith module of deficiency of an ideal/module
+--=======================================================================
+MyDoc=concatenate(MyDoc,///
+Node
+  Key
+    deficiencyModule
+    (deficiencyModule,Module,ZZ)
+    (deficiencyModule,Ideal,ZZ)
+  Headline
+    computes the $i$th module of deficiency of a module $M$ or an ideal $I$.
+  Usage
+    deficiencyModule(M,i)
+    deficiencyModule(I,i)
+  Inputs
+    M:Module
+      module over the polynomial ring $S=K[x_1,\ldots,x_n]$, with $K$ a field or an ideal $I\subset S$
+    i:ZZ
+      an integer
+  Outputs
+    D:Module
+      the ith module of deficiency of $M$
+  Description
+    Text
+      Let $S=K[x_1,\ldots,x_n]$ be the polynomial ring.
+      Given a $S$-module $M$ (or an ideal $I\subset S$), it returns the $i$th module of deficiency $\omega^i(M)$ (or $\omega^i(S/I)$), defined as $\mathrm{Ext}_S^{n-i}(M,S(-n))$.
+    Example
+      S=QQ[x_1..x_5];
+      I=(x_1^2*x_3,x_2*x_3^2*x_4,x_1*x_3^3*x_5);
+      M=S^1/I;
+      deficiencyModule(M,3)
+  SeeAlso
+    canonicalModule
+    isSCM
+///);
+-------------------------------------------------------------------------
+deficiencyModule = method(TypicalValue=>Module);
+-------------------------------------------------------------------------
+deficiencyModule(Module,ZZ) := (M,i) -> (
+    S:=ring M;
+    n:=dim S;
+    t:=depth M;
+    d:=dim M;
+    if i<t or i>d then (
+        Di:=Ext^(n+1)(M,S^{-n})
+    ) else Di=Ext^(n-i)(M,S^{-n});
+    return Di
```

Comment:
`return` on the last line of a function is unnecessary.  This line be simplied to just `Di`.

The same is true for many of the other functions in this package.